spec/solver_spec.rb

```diff
@@ -0,0 +1,22 @@
+require_relative '../solver'
+
+describe Solver do
+  solver = Solver.new
+  context ' Test the Solver class'
+  it 'test the factorial method' do
+    expect(solver.factorial(4)).to eq 24
+    expect(solver.factorial(5)).to eq 120
+  end
```

Comment:
- [ ] try to be as thorough as possible when testing, in this case, that means including a test when a negative number is provided and when 0 is provided.